packages/sitecore-jss-nextjs/src/editing/editing-data-cache.ts

```diff
@@ -27,22 +27,28 @@ export class EditingDataDiskCache implements EditingDataCache {
     this.cache = new Cache('editing-data', { compression: 'gzip', location: tmpDir });
   }
 
-  set(key: string, editingData: EditingData): void {
+  set(key: string, editingData: EditingData): Promise<void> {
     const filePath = this.cache.set(key, JSON.stringify(editingData));
-    if (!filePath || filePath.length === 0) {
-      throw new Error(`Editing data cache not set for key ${key} at ${this.cache.root}`);
-    }
+    return new Promise((resolve) => {
+      if (!filePath || filePath.length === 0) {
+        throw new Error(`Editing data cache not set for key ${key} at ${this.cache.root}`);
```

Comment:
Should we use `reject` here for good measure (even though it wouldn't matter in our current usage)?

Reply:
Yes, it makes sense to use reject here, throw new Error will also fall into a catch, but I think it's cleaner to use reject.